### ISSUE TITLE
OpenAndroMaps can't be downloaded on Android7

### ISSUE THREAD

**Issue by MakiWolf** [cgeo/cgeo#14563]
### Describe your problem!

OpenAndroMaps can't be downloaded on Android7 with Sony Xperia XA.
Only Elevate file can be downloaded.

Maybe it is blocked by browser user agent or something else? Which browser user agent does cgeo use on Android7?

![Screenshot_20230829-211322](https://github.com/cgeo/cgeo/assets/73300633/8e0dadc8-9f2b-4ac4-95d8-7c5ce0a05de3)

### How to reproduce?

settings
Map Sources
Download offline map
OpenAndroMaps

### Actual result after these steps?

Only Elevate file can be downloaded.

### Expected result after these steps?

Should also display other downloads.

### Reproducible

Yes

### c:geo Version

2023.08.29-7dc4f16 developer build

### System information

```text
## System information

c:geo version: 2023.08.29-7dc4f16 developer build

Device:
-------
- Device type: F3111 (F3111, Sony)
- Available processors: 8
- Android version: 7.0
- Android build: 33.3.A.1.126
- Screen resolution: 720x1184px (360x568dp)
- Pixel density: 2.0
- System font scale: 1.0 / used scale: 1.0
- Sailfish OS detected: false
- Google Play services: enabled - 23.31.16 (040400-555695945)
- Memory:  Available:718,5 MB, Total:1,8 GB, Threshold: 234,8 MB, low:false

Sensor and location:
-------
- Low power mode: inactive
- Compass capabilities: yes
- Rotation vector sensor: absent
- Orientation sensor: present
- Magnetometer & Accelerometer sensor: present
- Direction sensor used: magnetometer & accelerometer

Program settings:
-------
- Settings: v9, Count:86
- Set language: de_AT (system default)
- System date format: dd.MM.yy
- Time zone: GMT+02:00
- Debug mode active: no
- Log Settings: debug=false, minLevel=WARN, minAddCallerInfo=NONE, addCallerInfoMaxDepth=8, throwOnError=false, transactionSizes=false
- Last manual backup: never
- Last auto backup: never
- Routing mode: Walk
- Live map mode: true
- OSM multi-threading: true / threads: 4
- Map: Austria (Offline)
  - Id: cgeo.geocaching.maps.mapsforge.MapsforgeMapProvider$OfflineMapSource:primary:neugeo/maps/austria.map
  - Atts: Map data (c) OpenStreetMap contributors
  - Theme: none

Filters:
-------
- Hide waypoints: -
- LIVE: Keine ([:advanced=false:inconclusive=false])
- OFFLINE: Keine ([:advanced=false:inconclusive=false])

Stored Filters (#0):

Services:
-------
- Geocaching sites enabled:
   - geocaching.com: Logged in (Anmeldung OK) / BASIC
- Geocaching.com date format: MM/dd/yyyy
- Geocaching.com website language: en-US
- Last successful login on geocaching.com: 2023-08-29 21-12
- Routing: internal / BRouter installed: false
- Installed c:geo plugins:  none

Permissions
-------
- android.permission.ACCESS_COARSE_LOCATION:: granted
- android.permission.ACCESS_FINE_LOCATION:: granted
- android.permission.READ_CONTACTS:: DENIED
- android.permission.READ_EXTERNAL_STORAGE:: granted
- android.permission.WRITE_EXTERNAL_STORAGE:: granted

Paths
-------
- System internal c:geo dir: /data/user/0/cgeo.geocaching (5,8 GB free) v3 internal isDir(6 entries)
- Legacy User storage c:geo dir: /storage/emulated/0/cgeo (5,8 GB free) v3 external non-removable isDir(7 entries)
- Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (5,8 GB free) v3 external non-removable isDir(3 entries)
- Internal theme sync (is turned off): /data/user/0/cgeo.geocaching/MapThemeData (5,8 GB free) v3 internal isDir(0 entries)
- Map render theme path: 
- Public Folders: #12
  - BASE: /neugeo (User-Defined)[/neugeo[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3Aneugeo::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Aneugeo/document/primary%3Aneugeo, Av:true, files:>=2, dirs:>=9, totalFileSize:>=481,3 MB, free space: 5,8 GB, files on device: 679936)
  - OFFLINE_MAPS: /neugeo/maps (Default)[/neugeo/maps[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Aneugeo::/maps]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Aneugeo/document/primary%3Aneugeo%2Fmaps, Av:true, files:2, dirs:2, totalFileSize:481,3 MB, free space: 5,8 GB, files on device: 679936)
  - OFFLINE_MAP_THEMES: /neugeo/maps/_themes (Default)[/neugeo/maps/_themes[PERSISTABLE_FOLDER(OFFLINE_MAPS)#1:p-content://com.android.externalstorage.documents/tree/primary%3Aneugeo::/maps/_themes]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Aneugeo/document/primary%3Aneugeo%2Fmaps%2F_themes, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 5,8 GB, files on device: 679936)
  - OFFLINE_MAP_SHADING: /neugeo/maps/_hgt (Default)[/neugeo/maps/_hgt[PERSISTABLE_FOLDER(OFFLINE_MAPS)#1:p-content://com.android.externalstorage.documents/tree/primary%3Aneugeo::/maps/_hgt]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Aneugeo/document/primary%3Aneugeo%2Fmaps%2F_hgt, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 5,8 GB, files on device: 679936)
  - LOGFILES: /neugeo/logfiles (Default)[/neugeo/logfiles[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Aneugeo::/logfiles]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Aneugeo/document/primary%3Aneugeo%2Flogfiles, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 5,8 GB, files on device: 679936)
  - GPX: /neugeo/gpx (Default)[/neugeo/gpx[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Aneugeo::/gpx]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Aneugeo/document/primary%3Aneugeo%2Fgpx, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 5,8 GB, files on device: 679936)
  - BACKUP: /neugeo/backup (Default)[/neugeo/backup[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Aneugeo::/backup]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Aneugeo/document/primary%3Aneugeo%2Fbackup, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 5,8 GB, files on device: 679936)
  - FIELD_NOTES: /neugeo/field-notes (Default)[/neugeo/field-notes[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Aneugeo::/field-notes]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Aneugeo/document/primary%3Aneugeo%2Ffield-notes, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 5,8 GB, files on device: 679936)
  - SPOILER_IMAGES: /neugeo/GeocachePhotos (Default)[/neugeo/GeocachePhotos[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Aneugeo::/GeocachePhotos]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Aneugeo/document/primary%3Aneugeo%2FGeocachePhotos, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 5,8 GB, files on device: 679936)
  - ROUTING_BASE: /neugeo/routing (Default)[/neugeo/routing[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Aneugeo::/routing]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Aneugeo/document/primary%3Aneugeo%2Frouting, Av:true, files:7, dirs:1, totalFileSize:89,9 KB, free space: 5,8 GB, files on device: 679936)
  - ROUTING_TILES: /neugeo/routing/segments4 (Default)[/neugeo/routing/segments4[PERSISTABLE_FOLDER(ROUTING_BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Aneugeo::/routing/segments4]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Aneugeo/document/primary%3Aneugeo%2Frouting%2Fsegments4, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 5,8 GB, files on device: 679936)
  - TEST_FOLDER: [Legacy]/data/user/0/cgeo.geocaching/files/unittest (Default)[/data/user/0/cgeo.geocaching/files/unittest[FILE#1:p-file:///data/user/0/cgeo.geocaching/files::/unittest]] (Uri: file:///data/user/0/cgeo.geocaching/files/unittest, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 5,8 GB, files on device: -1)
- PersistedDocumentUris: #2
  - PROXIMITY_NOTIFICATION_FAR: null
  - PROXIMITY_NOTIFICATION_CLOSE: null
- Persisted Uri Permissions: #1
  - content://com.android.externalstorage.documents/tree/primary%3Aneugeo (17. Aug., 21:13):RW
- Database: /data/user/0/cgeo.geocaching/databases/data (v100, Size:236,0 KB) on system internal storage

--- End of system information ---
```

**Comment by MakiWolf** [comment]
Just found out there is an old issue where test where failing in https://github.com/cgeo/cgeo/issues/13237

**Comment by Lineflyer** [comment]
I suspect the problem to be, that your old phone is either missing up-to-date root certificates or not supporting TLS1.2. 
More and more providers are removing TLS1.1 support these days.

If such is the case, there is nothing we can do about it from our side.
You could try to run your device on a newer version (if either provided by the vendor or by using a custom OS).

Can you try to use your phones browser to download a map from openandromaps? Should result in same problem if my theory is correct.

**Comment by ztNFny** [comment]
Please test with Androids browser, not Chrome

**Comment by MakiWolf** [comment]
@ztNFny Also with firefox on Android it worked. Do you have a specific browser in mind? Also interesting which library for download does cgeo use?

**Comment by ztNFny** [comment]
Firefox has its own SSL implementation, it doesn't use the systems. You need to test using the system browser - which is called "Browser" on standard Android.

**Comment by MakiWolf** [comment]
> You need to test using the system browser - which is called "Browser" on standard Android.

Thanks for the info but on Sony Xperia XA there is no app called "Browser".

**Comment by bekuno** [comment]
Please create a logfile (Settings -> System -> create logfile) directly after calling the map download site.
Then check it or upload it here or sent to support.
I believe that it is the same error as in https://github.com/cgeo/cgeo/issues/13237 - `HANDSHAKE_FAILURE_ON_CLIENT_HELLO`.

**Comment by MakiWolf** [comment]
@bekuno you are right here `HANDSHAKE_FAILURE_ON_CLIENT_HELLO` is in the logfile.
[logcat-2023-08-30-22-23-03-2.txt](https://github.com/cgeo/cgeo/files/12479914/logcat-2023-08-30-22-23-03-2.txt)

**Comment by bekuno** [comment]
duplicate of **https://github.com/cgeo/cgeo/issues/13237**
no fix available

**Comment by fm-sys** [comment]
Hi @MakiWolf,

I know that this is a kind of inconvenient way how an issue gets "solved". Therefore a workaround for you would be to download the files manually (as you already have tested) and move them into the cgeo maps directory afterwards. A bit more effort but fortunately you don't download offline maps every day ;-)

Probably a real fix _is_ possible (see Firefox), but that would definitely not be an easy one...

**Comment by Lineflyer** [comment]
I guess it will be only a matter of time until the device will no longer connect to gc.com. 
Already right now there should be a similar issue if trying to connect to opencaching.de AFAIR.

**Comment by eddiemuc** [comment]
If this is becoming an issue for such old systems, would this be the time to think about removing support for SDK24?

**Comment by MakiWolf** [comment]
@eddiemuc cgeo have when downloading in play store Android 5.0 or higher. So removing support for Android 7 and directly jump to Android 8 or higher will be a big step. I also don't know how many users, use old devices.

**Comment by eddiemuc** [comment]
> @eddiemuc cgeo have when downloading in play store Android 5.0 or higher. So removing support for Android 7 and directly jump to Android 8 or higher will be a big step. I also don't know how many users, use old devices.

That is true. However, if we see that devices on such outdated OS versions are running out of up-to-date security standards then they won't be able to work with c:geo anyway. In that case maintaining these old versions would be in vain.

**Comment by moving-bits** [comment]
FYI: I've updated our [stats page](https://github.com/cgeo/cgeo/wiki/Playstore-statistics). According to Play Store stats from September 1st, 97.0% of recent c:geo users (last 30 days)) are using Android 8.0 or higher. This is a 1.0% increase compared to December.
